universal-login-react/src/ui/Notifications/ConnectionNotification.tsx

```diff
@@ -22,10 +22,15 @@ export const ConnectionNotification = ({deployedWallet, devicesBasePath, classNa
   const [showTitle, setShowTitle] = useState(true);
   const [gasParameters, setGasParameters] = useState<GasParameters | undefined>(undefined);
   const [publicKey, setPublicKey] = useState<string | undefined>(undefined);
-  useEffect(() => deployedWallet.subscribeAuthorisations(setNotifications), []);
 
   const history = useHistory();
 
+  const updateNotifications = (notifications: Notification[]) => notifications.length === 0
+    ? history.goBack()
+    : setNotifications(notifications);
+
+  useEffect(() => deployedWallet.subscribeAuthorisations(updateNotifications), []);
+
   const onDenyClick = async () => {
     await deployedWallet.denyRequests();
```

Comment:
Is `updateNotifications` callback executed after `await deployedWallet.denyRequests()`?
If so, then `history.goBack()` happens, and then we have `history.goBack()` again in the next line?

Reply:
Doesn't brake, but it is better for user experience. Done 👍